src/powfacpy/dyn_sim_interface.py

```diff
@@ -62,6 +62,129 @@ def create_event(self, name_incl_class, params={}, parent_folder=None, overwrite
     event_obj = self.create_in_folder(parent_folder, name_incl_class, overwrite=overwrite)
     self.set_attr(event_obj, params)  
 
+  def get_dsl_model_parameter_names(self, dsl_model):
+    """
+    Get the parameter names of the block definition (BlkDef)
+    of a dsl model.
+    """
+    dsl_model = self.handle_single_pf_object_or_path_input(dsl_model)
+    try: 
+      parameter_names = dsl_model.typ_id.sParams
+      if parameter_names:
+        return parameter_names[0].split(",")
+    except AttributeError:
+      msg = "Attribute 'typ_id' is of type 'None'"
+      raise powfacpy.PFObjectAttributeTypeError(dsl_model, msg, self)
+  
+  def get_dsl_models_inside_composite_model(self, composite_model):
+    return self.get_obj("*.ElmDsl", 
+                        parent_folder=composite_model,
+                        include_subfolders=True)
+  
+  def get_parameters_of_dsl_models_in_composite_model(
+      self, 
+      composite_model,
+      single_dict_for_all_dsl_models = False,
+      ):
+    """
+    Returns a dictionary with the parameter names (of the block definition) 
+    and values of all dsl models inside a composite model.
+    dsl lookup varibales (e.g. 'array_*', 'omatrix_*',.. ) are ignored.
+    
+    Arguments:
+      composite_model: ElmComp or its path
+      single_dict_for_all_dsl_models: 
+        - If true, a single dictionary with the parameters of all dsl models is 
+        returned (no distinction is made between the dsl models). 
+        This assumes that a parameter that occurs in several dsl
+        models has the same value.
```

Comment:
is this assumption valid? if not always, raise error if there are two parameters of same name but with different values.